command/catalog_nodes.go

```diff
@@ -0,0 +1,143 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/configutil"
+)
+
+// CatalogNodesCommand is a Command implementation that is used to fetch all the
+// datacenters the agent knows about.
+type CatalogNodesCommand struct {
+	BaseCommand
+}
+
+func (c *CatalogNodesCommand) Help() string {
+	helpText := `
+Usage: consul catalog datacenters [options]
+
+  Retrieves the list nodes registered in a given datacenter. By default, the
+  datacenter of the local agent is queried.
+
+  To retrieve the list of nodes:
+
+      $ consul catalog nodes
+
+  For a full list of options and examples, please see the Consul documentation.
+
+` + c.BaseCommand.Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *CatalogNodesCommand) Run(args []string) int {
```

Comment:
This is still a WIP, and I wanted to solicit feedback on the API/UX here.

I can see a few possible ways this could go:

- `consul catalog nodes` returns a list of node ids (no other data), and then `-detailed` returns a more tabular like (like `consul kv get -detailed`)
  - should it be node ids or node names?
  - should it be node names falling back to node ids?
- `consul catalog nodes` returns everything in tabular format except tagged addrs, meta, and create/modify indexes

Thoughts? I think I like the first one, since it allows for better composition into other resources.